### ISSUE TITLE
Problem with winrepo and pkg.installed

### ISSUE THREAD

**Issue by hrumph** [saltstack/salt#32962]
### Description of Issue/Question

With winrepo and using the "latest" package version it's hard to come up with a pkg.installed formula that works and I don't know how to come up with one that will allow me to turn off autoupdates

### Setup

First of all I'm using salt 2015.5.9 (both for the master and the minion). Apologies if this is fixed in a later version

Here is chrome.sls (simplified for testing purposes)

```
chrome:
  latest:
    full_name: 'Google Chrome'
    installer: 'https://dl.google.com/edgedl/chrome/install/GoogleChromeStandaloneEnterprise.msi'
    install_flags: '/qn /norestart'
    uninstaller: 'https://dl.google.com/edgedl/chrome/install/GoogleChromeStandaloneEnterprise.msi'
    uninstall_flags: '/qn /norestart'
    msiexec: True
    locale: en_US
    reboot: False
```

Here is a state formula for installing chrome.

```
chrome installed:
   pkg.installed:
     - name: chrome
```

If I apply the above formula, I get the following output

```
pkg_|-chrome installed_|-chrome_|-installed:
        ----------
        __run_num__:
            3
        changes:
            ----------
            chrome:
                ----------
                new:
                    66.153.16478
                old:
        comment:
            The following packages failed to install/update: chrome=latest
        duration:
            27298.0
        name:
            chrome
        result:
            False
        start_time:
            01:27:35.964000
```

So it appears to try to install "latest" but then it seems to complain possibly because the version that gets installed does not literally match "latest". For what it's worth the install does in reality succeed.

So far the only formula that works 100% for me is:

```
chrome installed:
   pkg.installed:
     - name: chrome
     - pkgs:
          - chrome
```

I think that this formula may work because it doesn't try any version match test when you do it this way. However with this formula I'm still unable to specifiy a specific version, so I have to use it in conjunction with Chrome auto-updates (not that auto-updates are a bad thing but
I'd like to have the choice). 

Is there a way around this? One way to handle this might be something like:

If the version number you ask for doesn't match anything in the winrepo sls file, then think of "latest" as a wildcard matching all the unmatching version numbers and then run the installer and check that the resulting version matches the asked for version.

I also tried using pkg.latest and couldn't get it to work (I had similar issues).

Thanks

**Comment by twangboy** [comment]
Support for `latest` in the software definition file was added in 2015.8. From a previous issue you submitted (#32961) I assume you're running 2015.5. Please test this with 2015.8.

**Comment by hrumph** [comment]
My bad I'm running 2015.5. Sorry

**Comment by jfindlay** [comment]
@hrumph, no problem and thanks for checking.

**Comment by hrumph** [comment]
OK, I updated the salt master to "2015.8.8.2 (Beryllium)".

I tried

```
chrome installed:
   pkg.installed:
     - name: chrome
```

and the result was

```
pkg_|-chrome installed_|-chrome_|-installed:
        ----------
        __run_num__:
            3
        changes:
            ----------
            chrome:
                ----------
                new:
                    66.153.16478
                old:
        comment:
            The following packages failed to install/update: chrome=latest
        duration:
            24938.0
        name:
            chrome
        result:
            False
        start_time:
            22:02:30.846000
```

It appears that even when it decides to go with "latest" it has to do a version check at the end and then matches the version installed with "latest". Perhaps the default behaviour, when "latest" is used should be to not do a version check at the end.

Note that even if you had another paramater for a target version to check against, it should really be a list of possible target versions, because consider that, even when Chrome has autoupdates disabled in registry settings, the version number changes a few moments after the installer exits!!! (you can see this in real time. It goes from 66.153.16478 to 50.0.2661.94). 

Anyway when Chrome is already installed obviously this configuration should do nothing (if you can't tell if you 've got the latest version or not because you haven't specified version numbers to compare with, then you should just do nothing), and that's what it does but it still compares with "latest" and returns false even so.

This may be OT but it's pretty clear that chrome really should be used with autoupdates enabled. The business of the changing version numbers in itself makes it totally unsuited to normal updating, so if salt were to just stop doing those comparisons I could have autoupdates on and everything would be fine.

As I said early on in this issue

```
chrome installed:
   pkg.installed:
     - name: chrome
     - pkgs:
          - chrome
```

works but that's just happenstance I guess.
There is the possibility that I'm just wrong about everything. If so can someone give me some pointers and maybe an example state formula for chrome installation? (I can't find anything like that anywhere on the web)

**Comment by twangboy** [comment]
@hrumph I'm sorry. My mistake. I just looked at the code and the PR I was referencing when I told you 2015.8 was cancelled. The PR actually went in to the develop branch and will be a part of the 2016.3 release.

**Comment by twangboy** [comment]
It's partially implemented in 2015.8 with this PR: https://github.com/saltstack/salt/pull/30800
The rest is in 2016.3 with this one which fixes it in the state module: https://github.com/saltstack/salt/pull/31114

**Comment by twangboy** [comment]
PR: #30800 made it into 2015.8.8... FYI, but it's only a partial fix. You need #31114 to fix the state module.

**Comment by hrumph** [comment]
Thanks, this can be closed for now I guess. I'm just going to take your word for it that this is fixed and I'll await the 2016.3 release.
